utils/docker/Dockerfile.el.9

```
@@ -0,0 +1,206 @@
+# Copyright 2022 Intel Corporation
+# All rights reserved.
+#
+# 'recipe' for Docker to build an image of EL 9 based
+# environment for building the DAOS project.
+#
+
+# Pull base image
+ARG BASE_DISTRO=almalinux:9
+FROM $BASE_DISTRO
+LABEL maintainer="daos@daos.groups.io"
+
+# Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
+ARG CB0
+
+# Use local repo server if present
+ARG REPO_FILE_URL
+RUN set -e;                                                      \
+    if [ -n "$REPO_FILE_URL" ]; then                             \
+        cd /etc/yum.repos.d/ &&                                  \
+        curl -k -f -o daos_ci-el9-artifactory.repo.tmp           \
+             "$REPO_FILE_URL"daos_ci-el9-artifactory.repo &&     \
+        for file in *.repo; do                                   \
+            true > $file;                                        \
+        done;                                                    \
+        mv daos_ci-el9-artifactory.repo{.tmp,};                  \
+    fi;                                                          \
+    dnf --disablerepo \*epel\* -y install epel-release           \
+                                          dnf-plugins-core;      \
+    if [ -n "$REPO_FILE_URL" ]; then                             \
+        PT_REPO=daos_ci-rocky8-crb-artifactory;                  \
+    else                                                         \
+        PT_REPO=crb;                                             \
+    fi;                                                          \
+    dnf -y config-manager --enable $PT_REPO;                     \
+    dnf -y clean all
```

There's a lot of code here that's supposed to interact with CI, however keeps being copied around between dockerfiles and I suspect is never tested.  I would prefer the following, and add back in the rest of the logic at a point where it's used rather than having untested code in the tree.
```suggestion
RUN set -e;                                                      \
    dnf --disablerepo \*epel\* -y install epel-release           \
                                          dnf-plugins-core;      \
    dnf -y config-manager --enable crb;                     \
    dnf -y clean all
```

---

> There's a lot of code here that's supposed to interact with CI, however keeps being copied around between dockerfiles and I suspect is never tested.

Because it is copied around and is the same, it is used and tested for every PR and landing in CI.  I would object to removing it as it's reasonable to assume that it will work and just becomes more work on the CI team to add it back in when somebody decides that it will be used in Jenkins.

> I would prefer the following, and add back in the rest of the logic at a point where it's used rather than having untested code in the tree.

-1 on that from me since that will be something that most likely lands on me to have to mop up at some point.  I would rather take the chance that it doesn't work an needs attention in the future than know that it absolutely will need attention at some point because it's entirely missing.

---

Brians choice on this one.  I prefer to start with a clean slate and copy in something known-working because I know this does suffer from bit-rot.

---

I would be less disagreeable if this were not more-or-less standard/generic boilerplate that is working in many other Dockerfiles and so has quite a reasonable probability of working in this one.  In fact, this, and other standard boilerplate functionality could quite conceivably be pulled into each of the Dockerfiles from a common source to alleviate any concerns of bitrot if somebody had the inclination.